geode-core/src/main/java/org/apache/geode/management/internal/configuration/domain/Configuration.java

```diff
@@ -183,20 +185,48 @@ public void fromData(DataInput in) throws IOException, ClassNotFoundException {
     propertiesFileName = DataSerializer.readString(in);
     gemfireProperties = DataSerializer.readProperties(in);
     HashSet<String> jarNames = DataSerializer.readHashSet(in);
-    if (jarNames != null) {
-      // we are reading pre 1.12 data. So add each jar name to deployments
-      jarNames.stream()
-          .map(x -> new Deployment(x, null, null))
-          .forEach(deployment -> deployments.put(deployment.getDeploymentName(), deployment));
-    } else {
+    try {
```

Comment:
Given that we now have a fromDataPre_X method, it should be unnecessary to have this try/catch block, because we can be sure that we'll be reading from a member with the correct version here.

Reply:
the catch part is when it is reading from a pre 1.12.0 version because it will hit the EOFException is trying to read the Configuration when it is being sent through EntryEventImpl from region (ClusterConfig region) replication.  EntryEventImpl does not send the version in the stream and hence the fromData_pre is not triggered.
But in the case of servers, they request it via function execution, version is sent through the stream and that's when the xxData_pre_XX is triggered.